pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,665 @@
+#include "lists.h"
+
+t_list   *allocate_list(size_t size, t_type type, void *elemnts)
+/*
+        Allocates a new t_list object
+
+        Parameters
+        ----------
+        size : size_t
+            the number of items in the list
+
+        type : t_type
+            The type of the new list's elements
+
+        elemnts : void*
+            A pointer on a table containing the list's elements
+
+        Returns
+        =======
+        list : t_list
+*/
+{
+    t_list *list;
+    size_t tsize = tSizes[type];
+
+    if (!(list = (t_list*)malloc(sizeof(t_list))))
+        return NULL;
+    list->size = size;
+    list->type = type;
+    list->capacity = DEFAULT_CAP;
+    while (list->capacity <= list->size)
+        list->capacity *= 2;
```

Comment:
Why `*=2`? And not `+= DEFAULT_CAP`? Are you not worried that this might lead to massive amounts of memory usage?

Reply:
I believe c++ vector container does the same, I think that a factor of 2 would minimize reallocations with the worst case scenario being 50% wasted memory. I also found this [https://stackoverflow.com/questions/1424826/why-is-vector-array-doubled](url)

Reply:
:+1: thanks. I didn't know that vector did this